packages/utils/src/big-lru-map.ts

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable max-classes-per-file */
+
+import LRUMap from 'mnemonist/lru-map';
+import { BigMap } from './big-map';
+
+/**
+* This used to ignore the init size of
+* the array in LRUMap. Doing this will avoid
+* running out of memory at 12 million records.
+* */
+export class FlexibleArray {
+    constructor() {
+        return [];
+    }
+}
+
+export class BigLRUMap<V> extends LRUMap<string|number, V> {
+    constructor(mapSize: number, keyArray?: FlexibleArray, valueArray?: FlexibleArray) {
```

Comment:
We need to make sure that the flexible array is passed in, also this should support a `TypedArray` (this comes from `@terascope/types`)

```suggestion
    constructor(mapSize: number, keyArray: FlexibleArray|TypedArray = FlexibleArray, valueArray: FlexibleArray|TypedArray = FlexibleArray) {
```

Reply:
If keyArray and valueArray are not optional would it make sense to move FlexibleArray to @terascope/types?

Reply:
Not really, this `FlexibleArray` is really a hack and isn't a type only thing. Ideally we shouldn't even be exporting it